server/routes/archiveGoal/archiveGoalController.test.ts

```diff
@@ -0,0 +1,258 @@
+      const expectedForm: ArchiveGoalForm = {
+        reference: goalReference,
+        title: goal.title,
+      }
+      const expectedView = {
+        prisonerSummary,
+        form: expectedForm,
+      }
+
+      // When
+      await controller.getArchiveGoalView(
+        req as undefined as Request,
+        res as undefined as Response,
+        next as undefined as NextFunction,
+      )
+
+      // Then
+      expect(res.render).toHaveBeenCalledWith('pages/goal/archive/reason', expectedView)
+      expect(req.session.archiveGoalForm).toBeUndefined()
+    })
+
+    it('should not get archive goal view given error getting prisoner action plan', async () => {
+      // Given
+      const actionPlan = { problemRetrievingData: true } as ActionPlan
+      educationAndWorkPlanService.getActionPlan.mockResolvedValue(actionPlan)
+
+      const expectedError = createError(500, `Error retrieving plan for prisoner ${prisonNumber}`)
+
+      // When
+      await controller.getArchiveGoalView(
+        req as undefined as Request,
+        res as undefined as Response,
+        next as undefined as NextFunction,
+      )
+
+      // Then
+      expect(next).toHaveBeenCalledWith(expectedError)
+      expect(req.session.archiveGoalForm).toBeUndefined()
+    })
+
+    it('should not get archive goal view given requested goal reference is not part of the prisoners action plan', async () => {
+      // Given
+      const someOtherGoalReference = 'd31d22bc-b9be-4d13-9e47-d633d6815454'
+      const goal = aValidGoal({ goalReference: someOtherGoalReference })
+      const actionPlan = aValidActionPlanWithOneGoal({ prisonNumber, goal })
+      educationAndWorkPlanService.getActionPlan.mockResolvedValue(actionPlan)
+
+      const expectedError = createError(404, `Goal ${goalReference} does not exist in the prisoner's plan`)
+
+      // When
+      await controller.getArchiveGoalView(
+        req as undefined as Request,
+        res as undefined as Response,
+        next as undefined as NextFunction,
+      )
+
+      // Then
+      expect(next).toHaveBeenCalledWith(expectedError)
+      expect(req.session.archiveGoalForm).toBeUndefined()
+    })
+  })
+  describe('submitArchiveGoalForm', () => {
+    it('should redirect to review archive goal page given action is submit-form and validation passes', async () => {
+      // Given
+      req.params.prisonNumber = 'A1234GC'
+
+      const archiveGoalForm = aValidArchiveGoalForm(goalReference)
+      req.body = { ...archiveGoalForm }
+
+      mockedValidateArchiveGoalForm.mockReturnValue([])
+
+      // When
+      await controller.submitArchiveGoalForm(
+        req as undefined as Request,
+        res as undefined as Response,
+        next as undefined as NextFunction,
+      )
+
+      // Then
+      expect(res.redirect).toHaveBeenCalledWith(`/plan/${prisonNumber}/goals/${goalReference}/archive/review`)
+      expect(mockedValidateArchiveGoalForm).toHaveBeenCalledWith(archiveGoalForm)
```

Comment:
Do we also need to assert that the form is on the session? (ie. so it can be used by the next controller method)
Is it worth adding an assertion like this (which I took from your next test scenario)
```
expect(req.session.archiveGoalForm).toEqual(expectedArchiveGoalForm)
```